doc: _extensions: scripts: apply ruff lint rules

kartben opened the pull request:
This makes all Python scripts in doc/ compliant with current ruff rules.

`redirects.py` is kept in the ignore list with regards to line length as keeping the ability to do block selection in this file can be particularly useful.

pdgendt commented:
#81070 has been merged 😉

kartben commented:
> #81070 has been merged 😉

will come back to this shortly then :) Thank you!

kartben committed:
doc: _extensions: apply ruff lint rules

This makes all Python scripts in doc/_extensions compliant w.r.t current
Ruff rules

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

pdgendt commented on doc/_extensions/zephyr/domain/__init__.py:
```diff
@@ -51,19 +53,14 @@
 sys.path.insert(0, str(Path(__file__).parents[3] / "_scripts"))
 
-from gen_boards_catalog import get_catalog
+from gen_boards_catalog import get_catalog  # noqa: E402
```
Is the `noqa` needed? I think import are allowed after `sys.path` calls.

kartben replied:
yes they are, but the `ZEPHYR_BASE = Path(__file__).parents[4]` is causing that 'algorithm' to fail. But I can always move things around, of course, it will just make the the paths used in the sys.path.insert more "obscure" thoughts?

pdgendt replied:
I think its fine to ignore for now then. I'm no expert in path resolution in python either, but I would think there'd be nicer ways. Having to do `sys.path.insert` in most files feels... icky.

kartben replied:
> I think its fine to ignore for now then. I'm no expert in path resolution in python either, but I would think there'd be nicer ways. Having to do `sys.path.insert` in most files feels... icky.

agreed, very icky :)
I've actually made the changes so that the noqa comment is not needed anymore. If anything, it makes the ickiness even more visible so that might encourage us to look for a better solution going forward :)

pdgendt commented on doc/_extensions/zephyr/kconfig/__init__.py:
```diff
@@ -62,13 +63,13 @@
-import list_boards
-import list_hardware
-import zephyr_module
+import kconfiglib  # noqa: E402
+import list_boards  # noqa: E402
+import list_hardware  # noqa: E402
+import zephyr_module  # noqa: E402
```
Same here

kartben replied:
same here :) it's the fact that there is more than just sys.path.insert() calls that's causing the check to not pass

kartben committed:
doc: _scripts: conf: apply ruff lint rules

This makes all remaining Python scripts in doc compliant w.r.t current
Ruff rules

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

pdgendt commented:
> `redirects.py` is kept in the ignore list with regards to line length as keeping the ability to do block selection in this file can be particularly useful.

I would add the ignore to the file instead of keeping it in the excludes file. Ideally the excludes file is removed at some point.

```python
# Ignore line-too-long to allow for block selection
# ruff: noqa: E501
```

See https://docs.astral.sh/ruff/tutorial/#ignoring-errors
